MdeModulePkg/Core/Dxe/DxeMain/DxeMain.c

```diff
@@ -261,15 +261,16 @@ DxeMain (
   //
   // Setup Stack Guard
   //
-  // MU_CHANGE START: Check Memory Protection HOB
+  // MU_CHANGE START: Check Memory Protection HOB, don't ASSERT if the exception stacks can't be created
   // if (PcdGetBool (PcdCpuStackGuard)) {
   GuidHob2 = GetFirstGuidHob (&gDxeMemoryProtectionSettingsGuid);
   if ((GuidHob2 != NULL) &&
       (((DXE_MEMORY_PROTECTION_SETTINGS *)GET_GUID_HOB_DATA (GuidHob2))->StructVersion == (UINT8)DXE_MEMORY_PROTECTION_SETTINGS_CURRENT_VERSION) &&
       ((DXE_MEMORY_PROTECTION_SETTINGS *)GET_GUID_HOB_DATA (GuidHob2))->CpuStackGuard)
   {
     Status = InitializeSeparateExceptionStacks (NULL, NULL);
-    ASSERT_EFI_ERROR (Status);
+    // ASSERT_EFI_ERROR (Status);

```

Comment:
This would be a major error on an x86 platform, right? Is it better to change the NULL lib to return EFI_SUCCESS, instead of removing this assert?

Reply:
It's a major issue if stack overflows or stack corruption issues are occurring. The InitializeSeparateExceptionStacks() routine is also used in the CPU Multiprocessor PEI and DXE instances where it definitely needs to know when the exception stack wasn't successfully created.

Reply:
If `EFI_UNSUPPORTED` is an expected failure. Can it still assert on other (unexpected) failures?

```suggestion
    ASSERT (Status == EFI_UNSUPPORTED || !EFI_ERROR (Status));
```

Reply:
Yes, I'll leave the print and use your version of the assert